spec/tracer_spec.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+# typed: false
+
+describe Redcord::Tracer do
+  class FakeTracer
+    def trace(*)
+      yield
+    end
+  end
+
+  context 'when tracer is set' do
+    include Redcord::Tracer::ClassMethods
+
+    let(:tracer) { proc { FakeTracer.new } }
+
+    it 'records traces' do
+      counter = 0
+
+      expect(
+        trace('test', model_name: 'test') { counter += 1 }
+      ).to be 1
+      expect(counter).to be 1
+    end
+  end
+
+  context 'when tracer is not set' do
+    include Redcord::Tracer::ClassMethods
+
+    let(:tracer) { nil }
+
+    it 'does not record traces' do
+      counter = 0
+      expect(
+        trace('test', model_name: 'test') { counter += 2 }
```

Comment:
(Don't want to block on unit tests or my lack of understanding here) Intuitively I would expect this to be 0 then, if it's not supposed to record traces?

Reply:
yeah its not recording traces, but we still need to run the block passed to this function.

Reply:
What in these tests is checking whether or not it's actually recording traces then?

Reply:
ooh i see why its confusing now. We should expect the fake tracer class to receive the `trace` method.

the other test case has it set to nil so it won't receive anything.